buoy_gazebo/src/MooringForce/MooringForce.cpp

```diff
@@ -292,28 +322,53 @@ void MooringForce::PreUpdate(
   // Unused at the moment
   double Tz = - this->dataPtr->w * (this->dataPtr->L - this->dataPtr->B[0U]);
 
-  gzdbg << "HSolver solverInfo: " << solverInfo
-    << " V: " << this->dataPtr->V
-    << " H: " << this->dataPtr->H
-    << " b: " << b
-    << " B: " << this->dataPtr->B[0U]
-    << " c: " << c
-    << " theta: " << this->dataPtr->theta
-    << " Tx: " << Tx
-    << " Ty: " << Ty
-    << " Tr: " << Tr
-    << " Tz: " << Tz
-    << std::endl;
+  // Throttle update rate
+  auto elapsed = _info.simTime - this->dataPtr->lastDebugPrintTime;
+  if (elapsed > std::chrono::steady_clock::duration::zero() &&
+      elapsed >= this->dataPtr->debugPrintPeriod)
+  {
+    this->dataPtr->lastDebugPrintTime = _info.simTime;
+    gzdbg << "HSolver solverInfo: " << solverInfo << "\n"
+      << " t: " << std::chrono::duration_cast<
+          std::chrono::milliseconds>(_info.simTime).count()/1000.0 << "\n"
+      << " Anchor: " << this->dataPtr->anchorPos << "\n"
+      << " Buoy:   " << this->dataPtr->buoyPos << "\n"
+      << " R: " << this->dataPtr->effectiveRadius << "\n"
+      << " L: " << this->dataPtr->L << "\n"
+      << " V: " << this->dataPtr->V << "\n"
+      << " H: " << this->dataPtr->H << "\n"
+      << " b: " << bMax << "\n"
+      << " B: " << this->dataPtr->B[0U] << "\n"
+      << " c: " << c << "\n"
+      << " theta: " << this->dataPtr->theta << "\n"
+      << " Tx: " << Tx << "\n"
+      << " Ty: " << Ty << "\n"
+      << " Tr: " << Tr << "\n"
+      << " Tz: " << Tz << "\n"
+      << " nfev: " << catenarySolver.nfev << "\n"
+      << " iter: " << catenarySolver.iter << "\n"
+      << " fnorm: " << catenarySolver.fnorm << "\n"
+      << "\n";
+  }
 
   // Did not find solution. Maybe shouldn't apply a force that doesn't make sense
   if (solverInfo != 1) {
+    gzerr << "HSolver failed to converge, solverInfo: " << solverInfo << "\n";
```

Comment:
A more explicit consequence would be helpful to a user (my suggestion is over 80 chars, feel free to edit)
```suggestion
    gzerr << "HSolver failed to converge, solverInfo: " << solverInfo << " No mooring force will be applied." << std::endl;
```

Reply:
LGTM.